### PR TITLE
Integrate electron-updater

### PR DESCRIPTION
**Pull request by nop33**
Related to #141

## Sound on 🔉 !

https://user-images.githubusercontent.com/1579899/201922192-92a6a8d8-e8eb-486f-88dd-b70287ad5ca9.mp4

Resources that helped me in this PR:
- https://www.electron.build/auto-update
- https://www.electron.build/configuration/publish
- https://mmelikes.medium.com/electron-auto-updater-with-frontend-manipulation-33d3bc5057f3
- https://www.electronjs.org/docs/latest/tutorial/ipc
- https://www.electron.build/auto-update#debugging

### CODE REVIEW

File: src/modals/UpdateWalletModal.tsx [hunk 1]
```diff
@@ -17,37 +17,96 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
...
 import CenteredModal, { ModalFooterButton, ModalFooterButtons } from './CenteredModal'
 
 interface UpdateWalletModalProps {
   onClose: () => void
   newVersion: string
+  startDownload?: boolean
 }
 
-const UpdateWalletModal = ({ onClose, newVersion }: UpdateWalletModalProps) => {
+type Status = 'download-available' | 'downloading' | 'download-finished' | 'download-failed'
```
Comment (mvaivre) [inline src/modals/UpdateWalletModal.tsx:35]: Nit: `UpdateStatus`?

File: src/modals/UpdateWalletModal.tsx [hunk 2]
```diff
@@ -17,37 +17,96 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
...
+  }, [])
+
+  useEffect(() => {
+    const removeListener = electron?.updater.onError((_, error) => {
+      setStatus('download-failed')
+      setError(error.toString())
+    })
+
+    return () => removeListener
+  }, [])
```
Comment (mvaivre) [inline src/modals/UpdateWalletModal.tsx:75]: Nit: could we regroup the 3 listeners setup in one `useEffect` call?

**Review by mvaivre — COMMENTED** [review]
Excellent. Just a couple of nits.

Also it seems like you didn't push your config changes in package.json, but I guess that's on purpose, as it's not useful until we're pointing to GH?

File: src/modals/UpdateWalletModal.tsx [hunk 3]
```diff
@@ -17,37 +17,96 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
...
+
+  useEffect(() => {
+    const removeListener = electron?.updater.onUpdateDownloaded(() => {
+      // Delay success message to give time for download validation errors to arise if any
+      setTimeout(() => setStatus('download-finished'), 1000)
+    })
+
+    return () => removeListener
```
Comment (mvaivre) [inline src/modals/UpdateWalletModal.tsx:65]: Is this really cleaning the effect up? 

To do so, here we would need to call `ipcRenderer.removeListener('updater:updateDownloaded', callback)`.

**Comment by nop33** [comment]
> Also it seems like you didn't push your config changes in package.json, but I guess that's on purpose, as it's not useful until we're pointing to GH?

This is correct. In fact, we shouldn't need any configs to be added because based on the docs, if the GitHub env var is detected, it uses GitHub as default. electron-updater is pretty smart ;)

> If GH_TOKEN or GITHUB_TOKEN is defined — defaults to [{provider: "github"}].

https://www.electron.build/configuration/publish

File: src/modals/UpdateWalletModal.tsx (hunk 3 above)
Comment (nop33) [reply src/modals/UpdateWalletModal.tsx:65]: You are absolutely right. Working on it.

Comment (nop33) [reply src/modals/UpdateWalletModal.tsx]: You were correct, thanks for spotting this! I tested my changes and everything seems to work. Pushed.

**Pull request merged**